Sources/AuthFoundation/Network/URLSessionProtocol.swift

```diff
@@ -17,21 +17,32 @@ import FoundationNetworking
 #endif
 
 /// Protocol defining the interface for interacting with a URLSession. This is used to provide mocking for unit tests.
+@_documentation(visibility: internal)
 public protocol URLSessionProtocol {
-    typealias DataTaskResult = (Data?, HTTPURLResponse?, Error?) -> Void
-    func dataTaskWithRequest(_ request: URLRequest, completionHandler: @escaping (Data?, URLResponse?, Error?) -> Void) -> URLSessionDataTaskProtocol
+    func data(for request: URLRequest) async throws -> (Data, URLResponse)
     var configuration: URLSessionConfiguration { get }
 }
 
-/// Protocol defining the interface for interacting with a URLSession. This is used to provide mocking for unit tests.
-public protocol URLSessionDataTaskProtocol {
-    func resume()
-}
-
+@_documentation(visibility: internal)
 extension URLSession: URLSessionProtocol {
-    @_documentation(visibility: internal)
-    public func dataTaskWithRequest(_ request: URLRequest, completionHandler: @escaping (Data?, URLResponse?, Error?) -> Void) -> URLSessionDataTaskProtocol {
-        dataTask(with: request, completionHandler: completionHandler)
+#if os(Linux)
```

Comment:
Do we tell people we support linux? Unless it's officially supported it would simplify our lives not to do this, who is testing linux?

Reply:
Yes we do. The support is marked as "Experimental", but its been a feature of the SDK since it was first released, and has been a good forcing-function to ensure the API structure is clean & abstract enough to allow developers to extend it as needed without requiring forking.